packages/backend-lalr/src/Happy/Backend/LALR/ProduceCode.lhs

```diff
@@ -233,61 +231,7 @@ reduction_1, ...   :: HappyReduction a b
 These are only generated if types for *all* rules are given (and not for array
 based parsers -- types aren't as important there).
 
->    produceTypes
->     | target == TargetArrayBased = id
-
->     | all isJust (elems nt_types) =
->       happyReductionDefinition . str "\n\n"
->     . interleave' ",\n "
->             [ mkActionName i | (i,_action') <- zip [ 0 :: Int .. ]
->                                                    (assocs action) ]
->     . str " :: " . str monad_context . str " => "
->     . intMaybeHash . str " -> " . happyReductionValue . str "\n\n"
->     . interleave' ",\n "
->             [ mkReduceFun i |
->                     (i,_action) <- zip [ n_starts :: Int .. ]
->                                        (drop n_starts prods) ]
->     . str " :: " . str monad_context . str " => "
->     . happyReductionValue . str "\n\n"
-
->     | otherwise = id
-
->       where tokens =
->               case lexer' of
->                       Nothing -> char '[' . token . str "] -> "
->                       Just _ -> id
->             happyReductionDefinition =
->                      str "{- to allow type-synonyms as our monads (likely\n"
->                    . str " - with explicitly-specified bind and return)\n"
->                    . str " - in Haskell98, it seems that with\n"
->                    . str " - /type M a = .../, then /(HappyReduction M)/\n"
->                    . str " - is not allowed.  But Happy is a\n"
->                    . str " - code-generator that can just substitute it.\n"
->                    . str "type HappyReduction m = "
->                    . happyReduction (str "m")
->                    . str "\n-}"
->             happyReductionValue =
->                      str "({-"
->                    . str "HappyReduction "
->                    . brack monad_tycon
->                    . str " = -}"
->                    . happyReduction (brack monad_tycon)
->                    . str ")"
->             happyReduction m =
->                      str "\n\t   "
->                    . intMaybeHash
->                    . str " \n\t-> " . token
->                    . str "\n\t-> HappyState "
->                    . token
->                    . str " (HappyStk HappyAbsSyn -> " . tokens . result
->                    . str ")\n\t"
->                    . str "-> [HappyState "
->                    . token
->                    . str " (HappyStk HappyAbsSyn -> " . tokens . result
->                    . str ")] \n\t-> HappyStk HappyAbsSyn \n\t-> "
->                    . tokens
->                    . result
->                 where result = m . str " HappyAbsSyn"
+>    produceTypes = id
```

Comment:
Looks like the entire function can be dropped now?

Reply:
That's right.